datafusion/functions-aggregate/src/first_last.rs

```diff
@@ -259,19 +260,17 @@ impl FirstValueAccumulator {
             })
             .collect::<Vec<_>>();
 
-        if self.ignore_nulls {
-            let indices = lexsort_to_indices(&sort_columns, None)?;
-            // If ignoring nulls, find the first non-null value.
-            for index in indices.iter().flatten() {
-                if !value.is_null(index as usize) {
-                    return Ok(Some(index as usize));
-                }
-            }
-            Ok(None)
+        let comparator = LexicographicalComparator::try_new(&sort_columns)?;
+
+        let min_index = if self.ignore_nulls {
+            (0..value.len())
+                .filter(|&index| !value.is_null(index))
+                .min_by(|&a, &b| comparator.compare(a, b))
```

Comment:
I double checked that `min_by` does not require the elements to be sorted already

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by